### PR TITLE
Add proper smearing for VASP

### LINKED ISSUE #434: `set_occupancy_smearing` should have `order` but not `ismear`
https://github.com/pyiron/pyiron_atomistics/blob/75c1edbc228606dd99e0384c2ed9c998e85ad768/pyiron_atomistics/vasp/base.py#L1285

During today's meeting we realised that it makes little sense to have `ismear` in `set_occupancy_smearing`, but it should have `order` (as I did in SPHInX).

### PR DESCRIPTION
**Pull request by sudarsan-surendralal**

### CODE REVIEW

File: pyiron_atomistics/vasp/base.py [hunk 1]
```diff
@@ -1280,21 +1280,29 @@ def set_electric_field(self, e_field=0.1, direction=2, dipole_center=None):
...
-    def set_occupancy_smearing(self, smearing="fermi", width=0.2, ismear=None):
+    @deprecate(arguments={"ismear": "Preferably use parameters `smearing` and `order` "
+                                    "to set the type of smearing you want"})
+    def set_occupancy_smearing(self, smearing: str = None, width: float = None,
+                               order: int = 1, ismear: int = None) -> None:
         """
         Set how the finite temperature smearing is applied in determining partial occupancies
 
         Args:
             smearing (str): Type of smearing (fermi/gaussian etc.)
```
Comment (samwaseda) [inline pyiron_atomistics/vasp/base.py]: I think it should explicitly say what's available.

File: pyiron_atomistics/vasp/base.py [hunk 2]
```diff
@@ -1280,21 +1280,29 @@ def set_electric_field(self, e_field=0.1, direction=2, dipole_center=None):
...
             self.input.incar["ISMEAR"] = int(ismear)
-        else:
-            self.input.incar["ISMEAR"] = ismear_dict[smearing]
-        self.input.incar["SIGMA"] = width
+        elif smearing.lower().startswith("meth") or smearing.lower().startswith("mp"):
+            self.input.incar["ISMEAR"] = int(order)
+        elif smearing.lower().startswith("fermi"):
+            self.input.incar["ISMEAR"] = -1
+        elif smearing.lower().startswith("gauss"):
+            self.input.incar["ISMEAR"] = 0
```
Comment (samwaseda) [inline pyiron_atomistics/vasp/base.py:1303]: Maybe `else ValueError`?

**Review by samwaseda — APPROVED** [review]
Thanks @sudarsan-surendralal !

File: pyiron_atomistics/vasp/base.py [hunk 3]
```diff
@@ -1280,21 +1280,29 @@ def set_electric_field(self, e_field=0.1, direction=2, dipole_center=None):
         if dipole_center is not None:
             self.input.incar["DIPOL"] = " ".join(str(val) for val in dipole_center)
 
-    def set_occupancy_smearing(self, smearing="fermi", width=0.2, ismear=None):
+    @deprecate(arguments={"ismear": "Preferably use parameters `smearing` and `order` "
+                                    "to set the type of smearing you want"})
```
Comment (pmrv) [inline pyiron_atomistics/vasp/base.py]:
```suggestion
    @deprecate(ismear="Preferably use parameters `smearing` and `order` "
                      "to set the type of smearing you want")
```

**Review by pmrv — APPROVED** [review]
Looks good to me except for the nits from me and @samwaseda .

**Pull request merged**